src/Components/Modal.js

```diff
@@ -3,54 +3,75 @@ import ReactDOM from 'react-dom';
 
 import '../styles/Modal.css'
 
-const Modal = ({
-  title,
-  poster,
-  backdrop,
-  averageRating,
-  releaseDate,
-  overview,
-  genres,
-  budget, 
-  revenue, 
-  tagline,
-  id,
-  selectedMovie,
-  isModalOpen,
-  toggleModal
-}) => {
-  const myStyle = {
-    backgroundImage: `url(${backdrop})`
+import { fetchSingleMovie } from '../api-Calls'
+
+class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movie: this.props.selectedMovie,
+      title: null,
+      poster: null,
+      backdrop: null,
+      averageRating: null,
+      releaseDate: null,
+      overview: null,
+      genres: null,
+      budget: null,
+      revenue: null,
+      tagline: null,
+      isModalLoading: true,
+    }
   }
 
-  if (isModalOpen && selectedMovie === id) {
-    return ReactDOM.createPortal(
+
+componentDidMount = () => {
+  fetchSingleMovie(this.state.movie)
```

Comment:
You should have access to `this.props.selectedMovie` here - so it would make more sense to simply use that prop instead of saving it to state to use it. If you were going to manipulate that value later within this component or its children, then it would make sense to save it to state. 

🚨NOTE:🚨
This functionality will probably change a bit when you add in router, so the above comment might not even apply anymore. The main takeaway is: always ask yourself it it's necessary to save something to state or if you can just read it from props.

Reply:
Thank you Cass! I've changed the movie's initial state to null and passed the `this.props.selectedMovie` to our fetch call since it's the entire reason as to why we passed it down as props.